libs-scala/crypto/src/main/scala/com/daml/crypto/MessageDigestPrototype.scala

```diff
@@ -12,7 +13,7 @@ import java.security.MessageDigest
  * https://bugs.openjdk.java.net/browse/JDK-7092821, similar to Guava's
  * workaround https://github.com/google/guava/issues/1197
  */
-final class MessageDigestPrototype(val algorithm: String) {
+final class MessageDigestPrototype(val algorithm: String) extends StrictLogging {
```

Comment:
What's the motivation for this logging as opposed to using `com.daml.logging.ContextualizedLogger` or `org.slf4j.Logger`

Reply:
Ah, it's sugar over creating `org.slf4j.Logger` by hand

Reply:
I was mainly looking how in general Loggers are created and found this is wide pattern. 

Do we have some rules or best practices for the codebase in regards to the logging?

Reply:
My recommendation would be to use the generally accepted good practice of favoring composition over inheritance, including in this case. Hence, I would recommend to instantiate your logger statically and use that. This is the approach wherever we use contextualized loggers.

If all constructor call sites have a logging context in place, you probably want to use contextualized logging.

Reply:
Almost none of constructor call sites are having logging context in place, so I go with just instantiating the logger statically. The usage of this class is so deeply nested within classes like `Hash` or `Utf8`, passing through the context looks like massive exercise and complicating so many method signatures.